File: arrays/transpose_matrix.py
class Solution:
    def transpose(self, matrix: List[List[int]]) -> List[List[int]]:
        m = len(matrix)
        n = len(matrix[0])
        
        t_matrix = [[0] * m for _ in range(n)]
        
        for i, row in enumerate(matrix):
            for j, col in enumerate(row):
                t_matrix[j][i] = col
        
        return t_matrix

File: arrays/transpose_matrix.rb
# @param {Integer[][]} matrix
# @return {Integer[][]}
def transpose(matrix)
    m = matrix.size
    n = matrix[0].size

    t_matrix = Array.new(n){Array.new(m)}

    matrix.each_with_index do |row, i|
        row.each_with_index do |col, j|
            t_matrix[j][i] = col
        end
    end

    t_matrix
end